spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -266,7 +267,17 @@ private boolean doPoll() {
 			if (holder != null) {
 				holder.setMessage(message);
 			}
-			this.handleMessage(message);
+			try {
+				this.handleMessage(message);
+			}
+			catch (Exception e) {
+				if (e instanceof MessagingException) {
+					throw new OriginalMessageContainingMessagingException(message, (MessagingException) e);
+				}
+				else {
+					throw new MessagingException(message, e);
+				}
```

Comment:
I am slightly confused. Why do we need a new class if existing one (`MessagingException`) provides everything you need to hold and access both the _cause_ and the _message_.
Is it just a Marker interface? If so you could still simplify its implementation since you can still rely on `getFailedMessage()` and other provided functionality 
What am I missing?